src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/PensionAnnexDateStampServiceTest.java

```diff
@@ -0,0 +1,131 @@
+
+    @InjectMocks
+    private PensionAnnexDateStampService service;
+    @Mock
+    private EvidenceManagementUploadService emUploadService;
+    @Mock
+    private EvidenceManagementDownloadService emDownloadService;
+    @Mock
+    private GenericDocumentService genericDocumentService;
+    private final String caseId = "123123123";
+    private final LocalDate approvalDate = LocalDate.of(2024, 12, 31);
+    @Captor
+    private ArgumentCaptor<List<MultipartFile>> filesCaptor;
+    private CaseDocument document;
+
+
+    @BeforeEach
+    void setUp() {
+        document = CaseDocument.builder()
+            .documentBinaryUrl("https:mockurl/binary")
+            .documentFilename("Testfile")
+            .documentUrl("http:mockfile").build();
+    }
+
+    @Test
+    void shouldAddApprovalDateToPensionOrderDocument() throws Exception {
+        byte[] docInBytes = loadResource("/fixtures/P1_pension_sharing_annex.pdf");
+        when(emDownloadService.download(document.getDocumentBinaryUrl(), AUTH_TOKEN))
+            .thenReturn(docInBytes);
+        when(emUploadService.upload(any(), anyString(), any()))
+            .thenReturn(fileUploadResponse());
+        when(genericDocumentService.toCaseDocument(any(Document.class))).thenCallRealMethod();
+        service.appendApprovedDateToDocument(document, AUTH_TOKEN, approvalDate, caseId);
+
+        verify(emUploadService).upload(filesCaptor.capture(), anyString(), anyString());
+        List<MultipartFile> uploadedMultipartFiles = filesCaptor.getValue();
+        byte[] bytes = uploadedMultipartFiles.get(0).getBytes();
+        verifyDateOfOrderField(bytes);
+    }
+
+    @Test
+    void shouldNotAddApprovalDateToPensionOrderDocumentIfApprovalDateIsMissing() {
+        Exception exception = assertThrows(Exception.class, () -> service.appendApprovedDateToDocument(document, AUTH_TOKEN, null, caseId));
+        assertEquals("Missing or Invalid Approved Date of Order.", exception.getMessage());
+    }
+
+    @Test
+    void shouldNotAddApprovalDateToPensionOrderFlattenedDocument() throws IOException {
+        byte[] docInBytes = loadResource("/fixtures/P1_pension_sharing_annex_flattened.pdf");
+        when(emDownloadService.download(document.getDocumentBinaryUrl(), AUTH_TOKEN))
+            .thenReturn(docInBytes);
+
+        Exception exception = assertThrows(Exception.class, () -> service.appendApprovedDateToDocument(document, AUTH_TOKEN, approvalDate, caseId));
+        assertEquals("Pension Order document PDF is flattened / not editable.", exception.getMessage());
+    }
+
+    private byte[] loadResource(String testPdf) throws IOException {
+        try (InputStream resourceAsStream = getClass().getResourceAsStream(testPdf)) {
+            assert resourceAsStream != null;
+            return resourceAsStream.readAllBytes();
+        }
+    }
+
+    private void verifyDateOfOrderField(byte[] bytes) throws IOException {
+        try (PDDocument document = Loader.loadPDF(bytes)) {
```

Comment:
SonarLint complains about "document" in the IDE:
Rename "document" which hides the field declared at line 51.